sw/device/lib/base/mmio.c

```diff
@@ -31,7 +31,7 @@ static ptrdiff_t misalignment32_of(uintptr_t addr) {
  * @param from_mmio if true, copy from MMIO to main memory. Otherwise, copy from
  * main memory to MMIO.
  */
-static void mmio_region_memcpy32(mmio_region_t base, uint32_t offset, void *buf,
+static void mmio_region_memcpy32(mmio_region_t base, uint32_t offset, char *buf,
```

Comment:
I believe the issue here is the buffer increment below:

```c
buf += realignment
```

One approach is to change the type in the function parameters as it is done in this PR. The other is to create an alias pointer with a concrete type (e.g. `uint8_t *`). 

@mcy, what what are your thoughts here, noting that this is a static function?

Reply:
I don't have particularly strong feelings, though `uint8_t` would be more consistent with other code.

Another approach is to do `buf = ((char*) buf) + realignment);` but really `void*` is not a useful type to have in internal interfaces.

Reply:
char isn't really signedness-agnostic; the signedness of char is a bit of a contentious point in C, and dependent on the platform. It's not an issue here, but FYI.